eldritch/events.py

```diff
@@ -2006,22 +2058,41 @@ def BinaryChoice(
 
 class ItemChoice(ChoiceEvent):
 
-  def __init__(self, character, prompt):
+  def __init__(self, character, prompt, decks=None, item_type=items.Item):
     self.character = character
     self._prompt = prompt
     self.choices = None
+    self.chosen = None
+    if decks is None:
+      decks = {'spells', 'common', 'unique', 'skills', 'allies'}
+    assert decks.issubset({'spells', 'common', 'unique', 'skills', 'allies'}), f"{decks} contains invalid deck"
+    # can one choose a skill in any circumstances?
+    self.decks = decks
+    self.item_type = item_type
 
   def resolve(self, state, choice=None):
+    if self.choices is None:
+      self.compute_choices(state)
```

Comment:
You should never have to call `compute_choices` - it is handled by the event loop.

Reply:
There were some combat tests that were breaking because the event was not getting resolved in the usual way. I've fixed the test and removed this workaround.